src/sonic-config-engine/sonic-cfggen

```diff
@@ -371,6 +373,12 @@ def main():
             hardware_data['DEVICE_METADATA']['localhost'].update(asic_id=asic_id)
         deep_update(data, hardware_data)
 
+    # look for globaldb.config at /usr/share/sonic/<platform>/
```

Comment:
What does `get_globaldb_config(..)` do? Can the search for `globaldb.config` be done outside this module and pass the info as  parameter to this module.

Reply:
To locate the globaldb.config, platform info is required, but the platform info is obtained through callng sonic-cfggen.
The globaldb.config file is key-value pairs text file, get_lobaldb_config() parses it and return them in json. There are multiple shell scripts need parse the file, I feel it is better to centralize them in one place inside sonic-cfggen like sonic-cfggen parses machine.conf and asic.conf. 
passing globaldb.config info to sonic-cfggen will lost benefit to use sonic-cfggen. If it is not proper to wrapper globaldb.config parsing logic inside sonic-cfggen, we can just hardcode the parsing logic in different places, I feel that is hard to maintain in future.

Reply:
A counter example from existing code is [here](https://github.com/Azure/sonic-buildimage/blob/master/files/image_config/config-setup/config-setup#L212) Please notice that the platform is available as an envvar in `/etc/sonic/sonic-evironment` file.`sonic-cfggen` is heavy weight process and it does delay the switch start time.